configure-ironic.sh

```diff
@@ -67,15 +63,31 @@ env | grep "^OS_" | tee -a /etc/ironic/ironic.extra
 mkdir -p /shared/html
 mkdir -p /shared/ironic_prometheus_exporter
 
-if [ "$USE_HTTP_BASIC" = "true" ]; then
+HTPASSWD_FILE=/etc/ironic/htpasswd
+if [ -n "${HTTP_BASIC_HTPASSWD}" ]; then
+    printf "%s\n" "${HTTP_BASIC_HTPASSWD}" >"${HTPASSWD_FILE}"
+fi
+set_http_basic_server_auth_strategy() {
+    local section=${1:-DEFAULT}
+    crudini --set /etc/ironic/ironic.conf ${section} auth_strategy http_basic
+    crudini --set /etc/ironic/ironic.conf ${section} http_basic_auth_user_file "${HTPASSWD_FILE}"
+}
+
+# Configure HTTP basic auth for ironic-api server
+if [ -f "${HTPASSWD_FILE}" ]; then
+    set_http_basic_server_auth_strategy
+fi
+
 
-	crudini --set /etc/ironic/ironic.conf DEFAULT auth_strategy http_basic
-	crudini --set /etc/ironic/ironic.conf DEFAULT http_basic_auth_user_file /shared/htpasswd-ironic
-	crudini --set /etc/ironic/ironic.conf json_rpc auth_strategy http_basic
-	crudini --set /etc/ironic/ironic.conf json_rpc http_basic_auth_user_file /shared/htpasswd-ironic
-	crudini --set /etc/ironic/ironic.conf json_rpc http_basic_username $IRONIC_HTTP_BASIC_USERNAME
-	crudini --set /etc/ironic/ironic.conf json_rpc http_basic_password $IRONIC_HTTP_BASIC_PASSWORD
+# Configure auth for clients
+IRONIC_API_CONFIG_OPTIONS="--config-file /usr/share/ironic/ironic-dist.conf --config-file /etc/ironic/ironic.conf"
 
-	## NOTE(iurygregory): reusing the ironic credentials so we don't end up with wrong client credentials
-	htpasswd -nbB $IRONIC_HTTP_BASIC_USERNAME $IRONIC_HTTP_BASIC_PASSWORD > /shared/htpasswd-ironic
-fi
+configure_client_basic_auth() {
+    local auth_config_file="/auth/$1/auth-config"
+    if [ -f ${auth_config_file} ]; then
+        IRONIC_API_CONFIG_OPTIONS+=" --config_file ${auth_config_file}"
+    fi
+}
+
+configure_client_basic_auth ironic-inspector
```

Comment:
Do we need to configure inronic-inspector and pass the config for the ironic-api?

Reply:
Basically this is configuring ironic-api to be a client of ironic-inspector, by adding the config file `/auth/ironic-inspector/auth-config` if it exists. This file will look something like:
```
[inspector]
auth_strategy=http_basic
username=...
password=...
```

There's a corresponding change in the ironic-inspector image to allow it to authenticate as a client to ironic-api.